lib/web_ui/lib/src/engine/dom.dart

```diff
@@ -50,41 +98,84 @@ extension DomWindowExtension on DomWindow {
     );
   }
 
+  @JS('fetch')
+  external JSPromise _fetch1(JSString url);
+  @JS('fetch')
+  external JSPromise _fetch2(JSString url, JSAny headers);
+
   // ignore: non_constant_identifier_names
   external DomURL get URL;
-  external bool dispatchEvent(DomEvent event);
-  external DomMediaQueryList matchMedia(String? query);
-  DomCSSStyleDeclaration getComputedStyle(DomElement elt,
-          [String? pseudoElt]) =>
-      js_util.callMethod(this, 'getComputedStyle', <Object>[
-        elt,
-        if (pseudoElt != null) pseudoElt
-      ]) as DomCSSStyleDeclaration;
+
+  @JS('dispatchEvent')
+  external JSBoolean _dispatchEvent(DomEvent event);
+  bool dispatchEvent(DomEvent event) => _dispatchEvent(event).toDart;
+
+  @JS('matchMedia')
+  external DomMediaQueryList _matchMedia(JSString? query);
+  DomMediaQueryList matchMedia(String? query) => _matchMedia(query?.toJS);
+
+  @JS('getComputedStyle')
+  external DomCSSStyleDeclaration _getComputedStyle1(DomElement elt);
+  @JS('getComputedStyle')
+  external DomCSSStyleDeclaration _getComputedStyle2(
+      DomElement elt, JSString pseudoElt);
+  DomCSSStyleDeclaration getComputedStyle(DomElement elt, [String? pseudoElt]) {
+    if (pseudoElt == null) {
+      return _getComputedStyle1(elt);
+    } else {
+      return _getComputedStyle2(elt, pseudoElt.toJS);
+    }
+  }
+
   external DomScreen? get screen;
-  external double requestAnimationFrame(DomRequestAnimationFrameCallback callback);
+
+  @JS('requestAnimationFrame')
+  external JSNumber _requestAnimationFrame(JSFunction callback);
+  double requestAnimationFrame(DomRequestAnimationFrameCallback callback) =>
+      _requestAnimationFrame(callback.toJS).toDart;
+
+  @JS('postMessage')
+  external JSVoid _postMessage1(JSAny message, JSString targetOrigin);
+  @JS('postMessage')
+  external JSVoid _postMessage2(
+      JSAny message, JSString targetOrigin, JSArray messagePorts);
   void postMessage(Object message, String targetOrigin,
-          [List<DomMessagePort>? messagePorts]) =>
-      js_util.callMethod(this, 'postMessage', <Object?>[
-        message,
-        targetOrigin,
-        if (messagePorts != null) js_util.jsify(messagePorts)
-      ]);
+      [List<DomMessagePort>? messagePorts]) {
+    if (messagePorts == null) {
```

Comment:
Yea, it will be quite a bit faster to use static interop on Dart2Wasm. However, we can switch to using optional arguments, but then tearoffs won't work as expected.

If the consensus is this is okay, then we can do that change in a follow on. Hopefully in the nearish future I can convince everyone to make tearoffs of external functions a static error.